scalaj-client/src/main/scala/endpoints/scalaj/client/Endpoints.scala

```diff
@@ -0,0 +1,38 @@
+package endpoints.scalaj.client
+
+import endpoints.algebra
+import endpoints.algebra.MuxRequest
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Endpoints extends algebra.Endpoints with Requests with Responses {
+
+  override type MuxEndpoint[A, B, Transport] = Nothing
+
+  override def endpoint[A, B](request: Request[A], response: Response[B]): Endpoint[A, B] = {
+    Endpoint(request, response)
+  }
+
+  override def muxEndpoint[Req <: MuxRequest, Resp, Transport](request: Request[Transport], response: Response[Transport]): MuxEndpoint[Req, Resp, Transport] =
+    throw new UnsupportedOperationException("Not implemented")
+
+
+  case class Endpoint[Req, Resp](request: Request[Req], response: Response[Resp]) {
+
+    /**
+      * This method just wraps a call in a Future and is not real async call
+      */
+    def callAsync(args: Req)(implicit ec: ExecutionContext): Future[Resp] =
+      Future {
+        callUnsafe(args)
+      }
```

Comment:
`blocking` “tells” the execution context that this block of code may take several seconds to complete. Then, the execution context can take advantage of this information to manage its thread pool better.

~~~ scala
Future {
  concurrent.blocking {
    callUnsafe(args)
  }
}
~~~